CodingTracker.DreamFXX/UserInput.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Spectre.Console;
+using System.Globalization;
+
+namespace CodingTracker.DreamFXX
+{
+    internal class UserInput
+    {
+        readonly Validation _dateTimeValidation = new();
+        private string? Date { get; set; } = string.Empty;
+        private string? StartTime { get; set; } = string.Empty;
+        private string? EndTime { get; set; } = string.Empty;
+
+        public string? GetDate()
+        {
+            this.Date = AnsiConsole.Ask<string>(
+                "[yellow]Please, enter the date of your session. Specify date in this exact format![/]\n-[green](dd-MM-yy)[/] -> ");
+            while (Date != null && !_dateTimeValidation.IsValidDate(Date))
+            {
+                AnsiConsole.MarkupLine("Invalid date! Expected format (mm-dd-yy)");
+                this.Date = Console.ReadLine();
+            }
+            return Date;
+        }
+
+        public string? GetStartTime()
+        {
+            this.StartTime = AnsiConsole.Ask<string>
+                ("[yellow]Enter the time your session started. Specify time in this format![/]\n-[green](hh:mm)[/] -> ");
+
+            while (!_dateTimeValidation.IsValidStartTime(this.StartTime))
+            {
+                AnsiConsole.MarkupLine("[red]Invalid time! Time must be in this format - (hh:mm):[/] ");
+                this.StartTime = Console.ReadLine();
+            }
+
+            return this.StartTime;
+        }
+
+        public string? GetEndTime()
+        {
+            this.EndTime = AnsiConsole.Ask<string>
+                ("[yellow]Enter the time your session ended. Specify time in this format![/]\n-[green](hh:mm)[/] -> ");
+            while (!_dateTimeValidation.IsValidStartTime(this.EndTime))
+            {
+                AnsiConsole.MarkupLine("[red]Invalid time! Time must be in this format - (hh:mm):[/] ");
+                this.EndTime = Console.ReadLine();
+            }
+
+            return this.EndTime;
+        }
+
+        public string GetDuration()
+        {
+            if (StartTime != null)
+            {
+                DateTime parsedStartTime = DateTime.ParseExact(StartTime, "HH:mm", CultureInfo.InvariantCulture, DateTimeStyles.None);
+                if (EndTime != null)
+                {
+                    DateTime parsedEndTime = DateTime.ParseExact(EndTime, "HH:mm", CultureInfo.InvariantCulture, DateTimeStyles.None);
+
+                    TimeSpan duration = parsedEndTime.Subtract(parsedStartTime);
+                    if (duration < TimeSpan.Zero)
+                    {
+                        duration += TimeSpan.FromDays(1);
+                    }
+                    return duration.ToString();
+                }
+            }
+
+            return GetDuration();
```

Comment:
❓ Is this not an infinite loop?

If StartTime or EndTime is null, then you recursively return this function again, and again, and again.